app/src/docs/_props/epam-promo/components/pickers/common.tsx

```diff
@@ -38,8 +38,15 @@ export const getDataSourceExamples = (ctx: PropSamplesCreationContext) => {
         || new LazyDataSource({
             api: (request, context) => {
                 const { search } = request;
-                const filter = search ? {} : { parentId: context?.parentId };
-                return ctx.demoApi.locations({ ...request, search, filter });
+                if (search && context.parentId) { // >1 level, search
```

Comment:
let's also turn on `flattenSearchResults` for lazy datasources in property explorers